datafusion/core/src/execution/context/mod.rs

```diff
@@ -305,10 +306,18 @@ impl SessionContext {
 
     /// Creates a new `SessionContext` using the provided [`SessionState`]
     pub fn new_with_state(state: SessionState) -> Self {
+        let state_ref = Arc::new(RwLock::new(state.clone()));
+        state
+            .schema_for_ref("datafusion.public.xx")
```

Comment:
It just a workaround to get `DynamicFileSchemaProvider`. I'll refactor it later.